src/rust/aggregator/src/data/user_history.rs

```diff
@@ -70,227 +70,166 @@ impl Data for UserHistory {
             .execute("SET TRANSACTION ISOLATION LEVEL SERIALIZABLE;")
             .await
             .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
+        struct TxnVersion {
+            txn_version: BigDecimal,
+        }
+        let max_txn_version = sqlx::query_as!(
+            TxnVersion,
+            "SELECT txn_version FROM aggregator.user_history_last_indexed_txn",
+        )
+        .fetch_optional(&mut transaction as &mut PgConnection)
+        .await
+        .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
+        let txnv_exists = max_txn_version.is_some();
+        let max_txn_version = max_txn_version
+            .unwrap_or(TxnVersion { txn_version: BigDecimal::zero() })
+            .txn_version;
         let fill_events = sqlx::query!(
             r#"
                 SELECT * FROM fill_events
-                WHERE NOT EXISTS (
-                    SELECT * FROM aggregator.aggregated_events
-                    WHERE fill_events.txn_version = aggregated_events.txn_version
-                    AND fill_events.event_idx = aggregated_events.event_idx
-                )
+                WHERE txn_version > $1
                 ORDER BY txn_version, event_idx
             "#,
+            max_txn_version
         )
         .fetch_all(&mut transaction as &mut PgConnection)
         .await
         .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
         let change_events = sqlx::query!(
             r#"
                 SELECT * FROM change_order_size_events
-                WHERE NOT EXISTS (
-                    SELECT * FROM aggregator.aggregated_events
-                    WHERE change_order_size_events.txn_version = aggregated_events.txn_version
-                    AND change_order_size_events.event_idx = aggregated_events.event_idx
-                )
+                WHERE txn_version > $1
                 ORDER BY txn_version, event_idx
             "#,
+            max_txn_version
         )
         .fetch_all(&mut transaction as &mut PgConnection)
         .await
         .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
-        let cancel_events = sqlx::query!(
+        sqlx::query!(
             r#"
-                SELECT * FROM cancel_order_events
-                WHERE NOT EXISTS (
-                    SELECT * FROM aggregator.aggregated_events
-                    WHERE cancel_order_events.txn_version = aggregated_events.txn_version
-                    AND cancel_order_events.event_idx = aggregated_events.event_idx
-                )
+                INSERT INTO aggregator.user_history_limit
+                SELECT
+                    market_id,
+                    order_id,
+                    user,
```

Comment:
`user` should be quoted since this is a keyword

Reply:
Note that I caught this by moving the SQL into a separate query file and autoformatting on save with `pgformatter`, which made this all caps

Reply:
Fixed in #563